### ISSUE TITLE
High CPU usage and latency on aarch64 ARMV8 build

### ISSUE THREAD

**Issue by andrewtabarez** [xianyi/OpenBLAS#4722]
Hello, I am running into high latency and constant high CPU  usage (80-95% ) when running a ARMv8 build on my Samsung Note 10. To test bench this I am running various matrix unit tests. The armv7 build runs fast and shows moderate CPU usage when compared to this armv8 build so I am wondering if I am missing a certain configuration during build. Below are benchmarks to compare the armv8 and armeabi-v7a builds running the exact same code. My best guess on the high CPU usage is some thread locking issue for some reason, though I don't have anything substantial to back that up.

BUILD INSTRUCTIONS I USED: 
**ARMV8 build (android-ndk-r18b-2022) (version 0.3.27.dev)**
make TARGET=ARMV8 \
    CC=$TOOLCHAIN_BIN/aarch64-linux-android-gcc \
    AR=$TOOLCHAIN_BIN/aarch64-linux-android-ar \
    FC=$TOOLCHAIN_BIN/aarch64-linux-android-gfortran \
    HOSTCC=gcc \
    -j4

(I have tried building with no threads / threads/ affinity on,off / locking / lots of threads / 1 thread) (these are defined in the Makefile.rule file, no difference here on CPU usage)

=================================================================
**armeabi-v7a build (android-ndk-r14b) (version 0.3.0.dev)**

make HOSTCC=gcc \
      CC="$COMPILERDIR/arm-linux-androideabi-gcc --sysroot=$SYSROOT" \
      FC="$COMPILERDIR/arm-linux-androideabi-gfortran --sysroot=$SYSROOT" \
      TARGET=ARMV7  \
      CROSS_SUFFIX=$COMPILERDIR/arm-linux-androideabi-gcc- \
      FCOMMON_OPT=-frecursive \
      ARM_SOFTFP_ABI=1 \

=================================================================
**BENCHMARKS** (Bolded are some tests that differ wildly between 64 and 32 bit builds) 
attached are android studio profile snippets showing that CPU usage is so much higher on the ARMv8 64bit builds)
**ARMv8**
64 bit
UnitTestRealFftSpeed,float,512,0.126857,seconds
UnitTestSplitRadixRealFftSpeed,float,512,0.0719979,seconds
**UnitTestSvdSpeed,float,4,39.363,seconds**
**UnitTestAddMatMatSpeed,float,2,1.87856,seconds**
UnitTestAddRowSumMatSpeed,float,5,0.228737,seconds
UnitTestAddColSumMatSpeed,float,5,0.30996,seconds
UnitTestAddVecToRowsSpeed,float,5,0.221573,seconds
UnitTestAddVecToColsSpeed,float,5,0.259116,seconds
UnitTestRealFftSpeed,double,512,0.140171,seconds
UnitTestSplitRadixRealFftSpeed,double,512,0.098691,seconds
**UnitTestSvdSpeed,double,4,32.5623,seconds**
**UnitTestAddMatMatSpeed,double,2,4.06978,seconds**
UnitTestAddRowSumMatSpeed,double,5,0.222114,seconds
UnitTestAddColSumMatSpeed,double,5,0.205234,seconds
UnitTestAddVecToRowsSpeed,double,5,0.240473,seconds
UnitTestAddVecToColsSpeed,double,5,0.162261,seconds
![armv8](https://github.com/OpenMathLib/OpenBLAS/assets/62033971/ee224307-e5da-403a-a862-8ec9066a7b3e)
As you can see the CPU usage is pinned above 90%.
**armeabi-v7a**
32bit
UnitTestRealFftSpeed,float,512,0.0684619,seconds
UnitTestSplitRadixRealFftSpeed,float,512,0.0363259,seconds
**UnitTestSvdSpeed,float,4,4.70356,seconds**
**UnitTestAddMatMatSpeed,float,2,0.76346,seconds**
UnitTestAddRowSumMatSpeed,float,5,0.140093,seconds
UnitTestAddColSumMatSpeed,float,5,0.142121,seconds
UnitTestAddVecToRowsSpeed,float,5,0.146741,seconds
UnitTestAddVecToColsSpeed,float,5,0.145406,seconds
UnitTestRealFftSpeed,double,512,0.0561302,seconds
UnitTestSplitRadixRealFftSpeed,double,512,0.0310419,seconds
**UnitTestSvdSpeed,double,4,7.92912,seconds**
**UnitTestAddMatMatSpeed,double,2,0.883866,seconds**
UnitTestAddRowSumMatSpeed,double,5,0.152569,seconds
UnitTestAddColSumMatSpeed,double,5,0.160446,seconds
UnitTestAddVecToRowsSpeed,double,5,0.148019,seconds
UnitTestAddVecToColsSpeed,double,5,0.147486,seconds
![armv7](https://github.com/OpenMathLib/OpenBLAS/assets/62033971/775b1b45-da56-4753-81ce-af45c310ec96)
CPU usage varies but usually under 50%

Any help here would be greatly appreciated! If more info is needed, I will gladly post it.

**Comment by martin-frbg** [comment]
Well, if you see essentially the same behaviour no matter if the build is multi- or singlethreaded, it is probably safe to rule out any locking issues. But I do wonder why you are comparing vastly different versions in addition to 32 vs 64 bit - 0.3.0 was released almost exactly six years ago and contained an equally old version of LAPACK (which may or may not matter, depending on what your benchmarks do). Did you build OpenBLAS yourself, or are these binaries you picked up from some other projects (which would imply differences in build options and compilers) ?
Do you see similar differences with the simple benchmarks provided in the benchmark directory ?

**Comment by andrewtabarez** [comment]
@martin-frbg thanks for the quick response.
I figured out this issue was a dumb mistake on my end. I was linking against this openblas aarch64 library in one of my libraries. Then when linking against that I was also accidentally pointing to the old armv7 32bit library of openblas. I would have thought that the compiler would complain here about linking 32bit and 64bit libs.....which I still do not understand.
Here is the CPU usage with the correct aarch64 lib. It's looking real nice now! Closing this issue
![image](https://github.com/OpenMathLib/OpenBLAS/assets/62033971/498e22c3-89fa-4dd6-9ddc-dbc5df9b2e49)
